op-batcher/batcher/batch_submitter.go

```diff
@@ -72,18 +73,16 @@ func Main(version string, cliCtx *cli.Context) error {
 		m.StartBalanceMetrics(ctx, l, batchSubmitter.L1Client, batchSubmitter.TxManager.From())
 	}
 
-	rpcCfg := cfg.RPCConfig
 	server := oprpc.NewServer(
-		rpcCfg.ListenAddr,
-		rpcCfg.ListenPort,
+		cfg.RPCFlag.ListenAddr,
+		cfg.RPCFlag.ListenPort,
 		version,
 		oprpc.WithLogger(l),
 	)
-	if rpcCfg.EnableAdmin {
-		server.AddAPI(gethrpc.API{
-			Namespace: "admin",
-			Service:   rpc.NewAdminAPI(batchSubmitter),
-		})
+	if cfg.RPCFlag.EnableAdmin {
+		rpcMetrics := opmetrics.NewRPCMetrics(procName, metrics.Namespace)
```

Comment:
I'd prefer that we used the top level metrics object. At the moment we still collect metrics even if they are disabled, but I'd like to move to not doing so. To enable that, it will be easier if we have a single instantiation of metrics rather metrics made per component.

Reply:
Gotcha. This makes total sense. Will update it to make it look similar to `opmetrics.RefMetrics`

Reply:
Updated here. I hope this is what you meant. Let me know if it's not. https://github.com/ethereum-optimism/optimism/pull/7482/commits/01af4459e933dbab220928aa35ec56b60f62a1c1